connectors/camunda-message/src/main/java/io/camunda/connector/message/SendMessageConnectorFunction.java

```diff
@@ -0,0 +1,147 @@
+        CorrelateMessageResponse correlateMessageResponse =
+            correlateMessageWithResponse(messageRequest);
+        LOG.debug(
+            "message correlated with message key {} and process instance key {}",
+            correlateMessageResponse.getMessageKey(),
+            correlateMessageResponse.getProcessInstanceKey());
+        return correlateMessageResponse;
+      }
+    }
+  }
+
+  private PublishMessageResponse publishMessageWithBuffer(
+      SendMessageRequest messageRequest, SendMessageRequest.CorrelationType.Publish publish) {
+    PublishMessageCommandStep3 publishMessageCommand;
+    PublishMessageCommandStep2 step2 =
+        zeebeClient.newPublishMessageCommand().messageName(messageRequest.messageName());
+    if (messageRequest.correlationKey() != null
+        && messageRequest.correlationKey().isBlank() == false) {
+      publishMessageCommand = step2.correlationKey(messageRequest.correlationKey());
+    } else {
+      publishMessageCommand = step2.withoutCorrelationKey();
+    }
+    if (publish.messageId() != null) {
+      publishMessageCommand.messageId(publish.messageId());
+    }
+    if (messageRequest.variables() != null) {
+      publishMessageCommand.variables(messageRequest.variables());
+    }
+    if (publish.timeToLive() != null) {
+      publishMessageCommand.timeToLive(publish.timeToLive());
+    }
+    if (messageRequest.tenantId() != null) {
+      publishMessageCommand.tenantId(messageRequest.tenantId());
+    }
+    if (messageRequest.requestTimeout() != null) {
+      publishMessageCommand.requestTimeout(messageRequest.requestTimeout());
+    }
+    PublishMessageResponse publishMessageResponse = publishMessageCommand.send().join();
+    return publishMessageResponse;
+  }
+
+  private CorrelateMessageResponse correlateMessageWithResponse(SendMessageRequest messageRequest) {
+    CorrelateMessageCommandStep2 correlateMessageCommand =
+        zeebeClient.newCorrelateMessageCommand().messageName(messageRequest.messageName());
+    CorrelateMessageCommandStep3 correlateMessageCommandStep3;
+    if (messageRequest.correlationKey() == null || messageRequest.correlationKey().isBlank()) {
+      correlateMessageCommandStep3 = correlateMessageCommand.withoutCorrelationKey();
+    } else {
+      correlateMessageCommandStep3 =
+          correlateMessageCommand.correlationKey(messageRequest.correlationKey());
+    }
+    if (messageRequest.variables() != null) {
+      correlateMessageCommandStep3.variables(messageRequest.variables());
+    }
+    if (messageRequest.tenantId() != null) {
+      correlateMessageCommandStep3.tenantId(messageRequest.tenantId());
+    }
+    if (messageRequest.requestTimeout() != null) {
+      correlateMessageCommandStep3.requestTimeout(messageRequest.requestTimeout());
+    }
+    CorrelateMessageResponse correlateMessageResponse = correlateMessageCommandStep3.send().join();
+    return correlateMessageResponse;
```

Comment:
nit: consider inlining variable

Reply:
Hi @chillleader, this is a leftover from a logging statement to print the result to the console. The compiler will inline it for us. But adding some log statements later on, will require a bigger refactoring then.